docs/spec/v1/buckets.md

```diff
@@ -662,22 +698,60 @@ The Provider allows for specifying the
 [Bucket location](https://cloud.google.com/storage/docs/locations) using the
 [`.spec.region` field](#region).
 
-##### GCP example
+##### GCP Controller-Level Workload Identity example
+
+```yaml
+---
+apiVersion: source.toolkit.fluxcd.io/v1
+kind: Bucket
+metadata:
+  name: gcp-controller-level-workload-identity
+  namespace: default
+spec:
+  interval: 5m0s
+  provider: gcp
+  bucketName: podinfo
+  endpoint: storage.googleapis.com
+  region: us-east-1
+  timeout: 30s
+```
+
+##### GCP Object-Level Workload Identity example
+
+**Note:** To use Object-Level Workload Identity (`.spec.serviceAccountName` with 
+cloud providers), the controller feature gate `ObjectLevelWorkloadIdentity` must 
+be enabled. Add the following environment variable to the source-controller 
+deployment:
+
+```yaml
+env:
+- name: ENABLE_OBJECT_LEVEL_WORKLOAD_IDENTITY
+  value: "true"
+```
```

Comment:
Let's remove this, the environment variable is set automatically by the controller `main()` when the feature gate is enabled (`auth.EnableObjectLevelWorkloadIdentity()`):

https://github.com/fluxcd/source-controller/blob/c2b572bae03a53cef4355389494ea71097c6abfe/main.go#L179-L185